src/ServerScriptService/Terrain/Generator/Support.lua

```diff
@@ -2,19 +2,47 @@ local Grid2D = require(script.Parent.Parent.Grid2D)
 
 local Support = {}
 
+function Support._getMinDistance(supportGrid, canadatePosition)
+	local minDistance = math.huge
+
+	supportGrid:IterateCells(function(position)
+		minDistance = math.min(minDistance, (canadatePosition - position).Magnitude)
+	end)
+
+	return minDistance
```

Comment:
This could be faster if it early outed, or optimally used a spatial partitioner - but it's good enough for now. TODO: open issue